### ISSUE TITLE
ParamSpec doesn't work when used in generic class

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1479]
PEP 612 shows an example of where a generic class can be parameterized using a ParamSpec. Pyright currently does not handle this correctly. Here's an example:

```python
from typing import Callable, Generic, TypeVar, ParamSpec


_P = ParamSpec("_P")
_R = TypeVar("_R")


class MyDecorator(Generic[_P, _R]):
    def __init__(self, function: Callable[_P, _R]):
        self.function = function

    def __call__(self, *args: _P.args, **kwargs: _P.kwargs) -> _R:
        print("Inside Function Call")
        return self.function(*args, **kwargs)


@MyDecorator  # Decorating Class
def my_func1(x: int, y: int):
    return x + y

my_func1(6, 6)
```

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.110, which I just published. It will also be included in the next release of Pylance.
